crates/starknet-server/src/api/http/endpoints/dump_load.rs

```diff
@@ -1,19 +1,38 @@
 use axum::{Extension, Json};
 
 use crate::api::http::error::HttpApiError;
-use crate::api::http::models::Path;
+use crate::api::http::models::{DumpLoadResponse, Path};
 use crate::api::http::{HttpApiHandler, HttpApiResult};
 
 pub(crate) async fn dump(
-    Json(_path): Json<Path>,
-    Extension(_state): Extension<HttpApiHandler>,
-) -> HttpApiResult<()> {
-    Err(HttpApiError::GeneralError)
+    Json(path): Json<Path>,
+    Extension(state): Extension<HttpApiHandler>,
+) -> HttpApiResult<Json<DumpLoadResponse>> {
+    if path.path.is_empty() {
+        return Err(HttpApiError::PathNotFound);
+    }
+    let starknet = state.api.starknet.write().await;
+    starknet
+        .dump_transactions_custom_path(Some(path.path.clone()))
+        .map_err(|_| HttpApiError::GeneralError)?;
+
+    Ok(Json(DumpLoadResponse { path: path.path }))
 }
 
 pub(crate) async fn load(
-    Json(_path): Json<Path>,
-    Extension(_state): Extension<HttpApiHandler>,
-) -> HttpApiResult<()> {
-    Err(HttpApiError::PathNotFound)
+    Json(path): Json<Path>,
+    Extension(state): Extension<HttpApiHandler>,
+) -> HttpApiResult<Json<DumpLoadResponse>> {
+    let file_path = std::path::Path::new(&path.path);
+    if path.path.is_empty() || !file_path.exists() {
```

Comment:
is `path.path` usual, can the name be different? just looks strange to me as a non-Rust person

Reply:
I think it's unavoidable since it's the local variable name of `struct Path` that contains `path` property that it's mapped from JSON. I could only change the local variable name to be `file` so it would be `file.path` but it's inappropriate and misleading in this case since there is no `File` in this context and it's just information about the path.